doc/md/versioned/03-auto-plan.md

```diff
@@ -0,0 +1,137 @@
+until this point. To demonstrate a more realistic use case, we will now switch to MySQL.
+See this change in [PR #3](https://github.com/rotemtam/ent-versioned-migrations-demo/pull/3/files).
+
+## Dev database
+
+To be able to plan accurate and consistent migration files, Atlas introduces the
+concept of a [Dev database](https://atlasgo.io/concepts/dev-database), a temporary
+database which is used to simulate the state of the database after different changes.
+Therefore, to use Atlas to automatically plan migrations, we need to supply a connection
+string to such a database to our migration planning script. Such a database is most commonly 
+spun up using a local Docker container. Let's do this now by running the following command:
+
+```shell
+docker run --rm --name atlas-db-dev -d -p 3306:3306 -e MYSQL_DATABASE=dev -e MYSQL_ROOT_PASSWORD=pass mysql:8
+```
+
+## Migration planning script
+
+Now that we have a Dev database, we can write a script that will use Atlas to plan
+migration files for us. Let's create a new file called `main.go` in the `ent/migrate` directory
+of our project:
+
+```go title=ent/migrate/main.go
+//go:build ignore
+
+package main
+
+import (
+    "context"
+    "log"
+    "os"
+    
+    // highlight-next-line
+    "<project>/ent/migrate"
+
+    atlas "ariga.io/atlas/sql/migrate"
+    "entgo.io/ent/dialect"
+    "entgo.io/ent/dialect/sql/schema"
+    _ "github.com/go-sql-driver/mysql"
+)
+
+func main() {
+    ctx := context.Background()
+    // Create a local migration directory able to understand Atlas migration file format for replay.
+    dir, err := atlas.NewLocalDir("ent/migrate/migrations")
+    if err != nil {
+        log.Fatalf("failed creating atlas migration directory: %v", err)
+    }
+    // Migrate diff options.
+    opts := []schema.MigrateOption{
+        schema.WithDir(dir),                         // provide migration directory
+        schema.WithMigrationMode(schema.ModeReplay), // provide migration mode
+        schema.WithDialect(dialect.MySQL),           // Ent dialect to use
+        schema.WithFormatter(atlas.DefaultFormatter),
+    }
+    if len(os.Args) != 2 {
+        log.Fatalln("migration name is required. Use: 'go run -mod=mod ent/migrate/main.go <name>'")
+    }
+    // Generate migrations using Atlas support for MySQL (note the Ent dialect option passed above).
+    //highlight-next-line
+    err = migrate.NamedDiff(ctx, "mysql://root:pass@localhost:3306/test", os.Args[1], opts...)
+    if err != nil {
+        log.Fatalf("failed generating migration file: %v", err)
+    }
+}
+```
+Notice that you need to make some modifications to the script above in the highlighted lines.
+Edit the import path of the `migrate` package to match your project and to supply the connection 
+string to your Dev database.
+
+To run the script, first create a `migrations` directory in the `ent/migrate` directory of your
+project:
+
+```text
+mkdir ent/migrate/migrations
+```
```

Comment:
Not sure what I think about it.

Reply:
I'll merge as is and we can discuss adding os.MkDirAll or similar